conductor/src/action.hpp

```diff
@@ -0,0 +1,192 @@
+
+#ifndef COCAINE_CONDUCTOR_ACTION_HPP
+#define COCAINE_CONDUCTOR_ACTION_HPP
+
+
+namespace cocaine { namespace isolate { namespace conductor {
+
+namespace action {
+
+struct cancellation_t:
+    public api::cancellation_t
+{
+    typedef std::shared_ptr<action_t> parent_ptr;
+    typedef std::weak_ptr<action_t> parent_wptr;
+
+    parent_wptr m_parent;
+
+    cancellation_t(parent_ptr parent);
+
+    virtual void cancel();
+};
+
+
+class action_t:
+    public enable_shared_from_this<action_t>
+{
+    // enum states {
+    //     st_pending = 0,
+    //     st_done,
+    //     st_cancelled
+    // };
+
+protected:
+
+    uint64_t             m_request_id;
+    shared_ptr<client_t> m_parent;
+
+    shared_ptr<io::basic_dispatch_t> m_dispatch;
+
+public:
+
+    struct states {
+        const static int st_pending = 0;
```

Comment:
Either these variables are used only in constant expressions or only once. Otherwise you'll get linker error someday.